### PR TITLE
Add hiddenWhen to debuggers in package.json

### PR DESCRIPTION
**Pull request by WardenGnaw**
This PR adds the 'hiddenWhen' attribute to debuggers contributed by the C# Extension.

Related: https://github.com/microsoft/vscode/pull/193157

Fixes: 
- https://github.com/microsoft/vscode-dotnettools/issues/14 
- https://github.com/microsoft/vscode-dotnettools/issues/512 
- https://github.com/microsoft/vscode-dotnettools/issues/513

Note: This PR requires the VS Code September 2023 Update

### CODE REVIEW

**Comment by dibarbet** [comment]
I assume the plan here is to wait for a 1.83 version of vscode to be released?

**Comment by timheuer** [comment]
> I assume the plan here is to wait for a 1.83 version of vscode to be released?

Do we have a pre-release train this could hop on for validation?

**Comment by dibarbet** [comment]
> > I assume the plan here is to wait for a 1.83 version of vscode to be released?
> 
> Do we have a pre-release train this could hop on for validation?

We don't currently have a planned prerelease, but we can ship a prerelease out of main at any time.  That prerelease would only work on vscode insiders though.

The other, more complicated problem is that our integration tests run on vscode release.  We would also have to switch the integration tests to acquire and run on insiders vscode in order to merge this PR (they're currently all failing on this PR for that reason)

**Comment by gregg-miskelly** [comment]
Do we need to wait? In other words, will older versions of VS Code just ignore the attribute?

**Comment by WardenGnaw** [comment]
These changes work if you use a version of vscode >1.83.0. We do not need the engine upgrade requirement.

File: package-lock.json [hunk 1]
```diff
@@ -108,7 +108,7 @@
         "webpack-cli": "4.6.0"
       },
       "engines": {
-        "vscode": "^1.73.0"
+        "vscode": "^1.83.0"
```
Comment (gregg-miskelly) [inline package-lock.json:111]: You want to undo this file

**Review by dibarbet — APPROVED** [review]

**Review by gregg-miskelly — APPROVED** [review]

**Pull request merged**